react-app/src/components/map/MapPropertyDetails.tsx

```diff
@@ -0,0 +1,278 @@
+          <Grid
+            item
+            xs={12}
+            sx={{
+              display: 'flex',
+              alignItems: 'center',
+            }}
+          >
+            <Button
+              variant="contained"
+              onClick={() => {
+                navigate(
+                  `/properties/${property.type === PropertyTypes.BUILDING ? 'building' : 'parcel'}/${property.id}`,
+                );
+              }}
+              sx={{
+                width: '100%',
+              }}
+            >
+              View Full Details
+            </Button>
+            <IconButton
+              sx={{
+                marginLeft: '0.5em',
+              }}
+              onClick={() => setOpen(false)}
+            >
+              <Close />
+            </IconButton>
+          </Grid>
+          <DividerGrid />
+          {isLoading ? (
+            <CircularProgress />
+          ) : (
+            <DrawerContents
+              property={property}
+              afterPropertyLoad={() => {
+                setOpen(true);
+                setIsLoading(false);
+              }}
+            />
+          )}
+        </Grid>
+      </Box>
+    </Drawer>
+  );
+};
+
+export interface ContentsProps extends MapPropertyDetailsProps {
+  afterPropertyLoad: () => void;
+}
+
+const DrawerContents = (props: ContentsProps) => {
+  const { property, afterPropertyLoad } = props;
+  const [propertyData, setPropertyData] = useState(undefined);
+  const [parcelLayerData, setParcelLayerData] = useState<ParcelData>(undefined);
+  const api = usePimsApi();
+
+  useEffect(() => {
+    if (property.id != null && property.type != null) getPropertyData();
+  }, [property]);
+
+  const getPropertyData = async () => {
+    let returnedProperty: Parcel | Building;
+    if (property.type === PropertyTypes.BUILDING) {
+      returnedProperty = await api.buildings.getBuildingById(property.id);
+    } else {
+      returnedProperty = await api.parcels.getParcelById(property.id);
+    }
+    setPropertyData(returnedProperty);
+    const latlng: LatLng = L.latLng(returnedProperty.Location.y, returnedProperty.Location.x);
+    api.parcelLayer.getParcelByLatLng(latlng).then((response) => {
+      if (response.features.length) {
+        setParcelLayerData(response.features.at(0).properties as ParcelData);
+      } else {
+        setParcelLayerData(undefined);
+      }
+    });
+    afterPropertyLoad();
+  };
+
+  return (
+    <>
+      {/* PARCEL INFO SECTION */}
+      <Grid item xs={12}>
+        <Typography variant="h4">{`Property Info`}</Typography>
+      </Grid>
+      {propertyData?.PID ? (
+        <>
+          <LeftGridColumn>PID</LeftGridColumn>
+          <RightGridColumn>{pidFormatter(propertyData?.PID)}</RightGridColumn>
+        </>
+      ) : (
+        <>
+          <LeftGridColumn>PIN</LeftGridColumn>
+          <RightGridColumn>{propertyData?.PIN}</RightGridColumn>
+        </>
+      )}
+      <LeftGridColumn>Name</LeftGridColumn>
```

Comment:
It could be worth combining the LeftGridColumn and RightGridColumn into one component since they pretty much always come in pairs as far as I can tell.

You could take it a step further and store the properties you want to display here in an object, then just enumerate the object and create a LeftRightGridColumn component for each entry. Manually specifying each component is fine too though, not like it really hurts performance or anything.

Reply:
Joined them into a single component where possible. 
Also addressed the issue with opening the pages in new tabs instead.
Hoping to see how the supercluster ticket works out, then will try to address they cluster behaviour and lag when components rerender.